### PR TITLE
Chronograf template variables with Flux

### LINKED ISSUE #2345: Enrich dashboard variables to include examples for Flux-based queries
Currently the dashboard template variables are shown being used in InfluxQL-based queries only.
There are no example Flux queries shown.  

Flux doesn't require these dashboard template variables to use the `:` syntax.  Also, `:interval:` is replaced by `autoInterval`.

##### Relevant URLs
https://docs.influxdata.com/chronograf/v1.8/guides/dashboard-template-variables/

### LINKED ISSUE #2724: Define Chronograf dashboard variables with Flux
**PR:** https://github.com/influxdata/chronograf/pull/5697

### PR DESCRIPTION
**Pull request by sanderson**
Closes #2345 
Closes #2724 

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

**Review by timhallinflux — CHANGES_REQUESTED** [review]

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 1]
```diff
@@ -144,12 +188,65 @@ In cell queries, it should be used in the `GROUP BY time()` clause that accompan
...
+which does not include "now", use `v.timeRangeStart` as the lower limit and
+[`v.timeRangeStop`](#vtimerangestop) as the upper limit.
+{{% /note %}}
+
+### v.timeRangeStop
+The `v.timeRangeStop` template variable is defined by the upper time limit specified using the date picker.
+
+<img src="/img/chronograf/1-6-template-vars-date-picker.png" style="width:100%;max-width:762px;" alt="Dashboard date picker"/>
+
+It will inherit `now()` when using relative time frames or the upper time limit when using absolute time frames.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:225]:
```suggestion
For relative time frames, this variable inherits `now()`. For absolute time frames, this variable inherits the upper time limit.
```

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 2]
```diff
@@ -90,14 +99,49 @@ SELECT mean("usage_user") AS "mean_usage_user" FROM "telegraf"."autogen"."cpu"
...
+
+Would display as:
+
+{{< img-hd src= "/img/chronograf/1-9-template-var-title.png" alt="Use template variables in cell titles" />}}
+
 ## Predefined template variables
 
-Chronograf includes predefined template variables controlled by elements in the Chrongraf UI.
+Chronograf includes predefined template variables controlled by elements in the Chronograf UI.
 These template variables can be used in any of your cells' queries.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:132]:
```suggestion
Use predefined template variables in your cell queries.
```

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 3]
```diff
@@ -144,12 +188,65 @@ In cell queries, it should be used in the `GROUP BY time()` clause that accompan
...
+```js
+from(bucket: "telegraf/autogen")
+  |> range(start: v.timeRangeStart)
+  |> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_system")
 ```
 
+{{% note %}}
+To use the date picker to specify a particular time range in the past
+which does not include "now", use `v.timeRangeStart` as the lower limit and
+[`v.timeRangeStop`](#vtimerangestop) as the upper limit.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:217]:
```suggestion
To use the date picker to specify a time range in the past
without "now", use `v.timeRangeStart` as the lower limit and
[`v.timeRangeStop`](#vtimerangestop) as the upper limit.
```

File: content/chronograf/v1.9/guides/dashboard-template-variables.md (hunk 1 above)
Comment (kelseiv) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:225]:
```suggestion
For relative time frames, this variable inherits `now()`. For absolute time frames, this variable inherits the upper time limit.
```

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 4]
```diff
@@ -144,12 +188,65 @@ In cell queries, it should be used in the `GROUP BY time()` clause that accompan
...
+from(bucket: "telegraf/autogen")
+  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
+  |> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_system")
+```
+
+### v.windowPeriod
+The `v.windowPeriod` template variable is controlled by the display width of the
+dashboard cell and is calculated by the duration of time that each pixel covers.
+Use the `v.windowPeriod` variable to downsample data to use only enough points to
+display one point per pixel.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:237]: To be sure, is this the sole use case?

Comment (kelseiv) [reply content/chronograf/v1.9/guides/dashboard-template-variables.md:237]: Is this accurate:
Use the `v.windowPeriod` variable to limit downsample data to display a maximum of one point per pixel.

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 5]
```diff
@@ -360,8 +458,16 @@ _**Example custom meta query variable in a cell query**_
 SELECT "purchases" FROM "animals"."autogen"."customers" WHERE "customer" = :customMetaVar:
 ```
 
-#### Custom meta query variable use cases
-Custom meta query template variables should be used any time you are pulling values from InfluxDB, but the pre-canned template variable types aren't able to return the desired list of values.
+#### InfluxQL meta query variable use cases
+Custom InfluxQL meta query template variables should be used any time you are pulling values from InfluxDB, but the pre-canned template variable types aren't able to return the desired list of values.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:462]:
```suggestion
Use custom InfluxQL meta query template variables when predefined template variable types aren't able to return the values you want.
```

Comment (kelseiv) [reply content/chronograf/v1.9/guides/dashboard-template-variables.md:462]: Curious if it makes sense to provide more detail here...or is it obvious when variables won't return values you want?

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 6]
```diff
@@ -360,8 +458,16 @@ _**Example custom meta query variable in a cell query**_
...
+
+### Flux Query
+Flux query template variables let you define variable values using Flux queries.
+**Variable values are extracted from the `_value` column returned by your Flux query.**
+
+#### Flux query variable use cases
+Flux query template variables are great when the values necessary for your
+variable can't be queried with InfluxQL.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:470]: same

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 7]
```diff
@@ -379,42 +485,60 @@ as well as many other parameters that control the display of graphs in your dash
...
+- `:lower:`
+- `:zoomedUpper:`
+- `:zoomedLower:`
+- `:refreshRate:`
 
 ## Advanced template variable usage
 
 ### Filter template variables with other template variables
-[Custom meta query template variables](#custom-meta-query) allow you to filter the array of potential variable values using other existing template variables.
+[Custom InfluxQL meta query template variables](#influxQL-meta-query) allow you to filter the array of potential variable values using other existing template variables.
```
Comment (kelseiv) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:502]:
```suggestion
[Custom InfluxQL meta query template variables](#influxQL-meta-query) let you filter the array of potential variable values using other existing template variables.
```

**Review by kelseiv — APPROVED** [review]

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 8]
```diff
@@ -90,14 +99,49 @@ SELECT mean("usage_user") AS "mean_usage_user" FROM "telegraf"."autogen"."cpu"
...
+
+{{< tabs-wrapper >}}
+{{% tabs %}}
+[InfluxQL](#)
+[Flux](#)
+{{% /tabs %}}
+{{% tab-content %}}
+- [`:dashboardTime:`](#dashboardtime)
+- [`:upperDashboardTime:`](#upperdashboardtime)
+- [`:autoInterval:`](#autointerval)
```
Comment (sanderson) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:144]:
```suggestion
- [`:interval:`](#interval)
```

File: content/chronograf/v1.9/guides/dashboard-template-variables.md (hunk 4 above)
Comment (sanderson) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:237]:
```suggestion
Use the `v.windowPeriod` variable to limit downsample data to display a maximum of one point per pixel.
```

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 9]
```diff
@@ -379,42 +485,60 @@ as well as many other parameters that control the display of graphs in your dash
...
-`:upper:`  
-`:lower:`  
-`:zoomedUpper:`  
-`:zoomedLower:`
-`refreshRate:`
+- `:database:`
+- `:measurement:`
+- `:dashboardTime:`
+- `:upperDashboardTime:`
+- `:autoInterval:`
```
Comment (sanderson) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:492]:
```suggestion
- `:interval:`
```

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 10]
```diff
@@ -132,9 +175,9 @@ FROM "telegraf".."cpu"
 WHERE time > :dashboardTime: AND time < :upperDashboardTime:
 ```
 
-### interval
+### autoInterval
```
Comment (sanderson) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:178]:
```suggestion
### interval
```

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 11]
```diff
@@ -132,9 +175,9 @@ FROM "telegraf".."cpu"
...
 
-The `:interval:` template variable is defined by the interval dropdown in the Chronograf dashboard.
+The `:autoInterval:` template variable is defined by the interval dropdown in the Chronograf dashboard.
```
Comment (sanderson) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:180]:
```suggestion
The `:interval:` template variable is defined by the interval dropdown in the Chronograf dashboard.
```

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 12]
```diff
@@ -144,12 +187,63 @@ In cell queries, it should be used in the `GROUP BY time()` clause that accompan
 SELECT mean("usage_system") AS "Average System CPU Usage"
 FROM "telegraf".."cpu"
 WHERE time > :dashboardtime:
-GROUP BY time(:interval:)
+GROUP BY time(:autoInterval:)
```
Comment (sanderson) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:190]:
```suggestion
GROUP BY time(:interval:)
```

File: content/chronograf/v1.9/guides/dashboard-template-variables.md [hunk 13]
```diff
@@ -360,8 +455,16 @@ _**Example custom meta query variable in a cell query**_
...
+#### InfluxQL meta query variable use cases
+Use custom InfluxQL meta query template variables when predefined template variable types aren't able to return the values you want.
+
+### Flux Query
+Flux query template variables let you define variable values using Flux queries.
+**Variable values are extracted from the `_value` column returned by your Flux query.**
+
+#### Flux query variable use cases
+Flux query template variables are great when the values necessary for your
+variable can't be queried with InfluxQL.
```
Comment (sanderson) [inline content/chronograf/v1.9/guides/dashboard-template-variables.md:467]:
```suggestion
Flux query template variables are great when the values necessary for your
variable can't be queried with InfluxQL or if you need the flexibility of Flux
to return your desired list of variable values.
```

**Pull request merged**